src/dev/Web/EastSeat.ResourceIdea.Web/Components/Pages/WorkItems/AddWorkItem.razor.cs

```diff
@@ -0,0 +1,285 @@
+                             HasAnyEngagements;
+
+    protected override async Task OnInitializedAsync()
+    {
+        // Parse query parameters to determine navigation source and context
+        var uri = NavigationManager.ToAbsoluteUri(NavigationManager.Uri);
+        var queryParams = QueryHelpers.ParseQuery(uri.Query);
+        
+        if (queryParams.TryGetValue("from", out var fromValue))
+        {
+            NavigationSource = fromValue.ToString();
+        }
+        else
+        {
+            // Determine navigation source based on query parameters
+            if (ClientId.HasValue && EngagementId.HasValue)
+            {
+                NavigationSource = "workitems";
+            }
+            else if (ClientId.HasValue)
+            {
+                NavigationSource = "client";
+            }
+            else if (EngagementId.HasValue)
+            {
+                NavigationSource = "engagement";
+            }
+            else
+            {
+                NavigationSource = "workitems";
+            }
+        }
+
+        await LoadEngagements();
+        await LoadEmployees();
+        
+        if (IsEngagementPreSelected)
+        {
+            SelectedEngagementId = EngagementId!.Value.ToString();
+        }
+        
+        IsLoading = false;
+        StateHasChanged();
+    }
+
+    private async Task LoadEngagements()
+    {
+        try
+        {
+            if (ClientId.HasValue)
+            {
+                // Load engagements for specific client
+                var clientQuery = new GetEngagementsByClientQuery(1, 100)
+                {
+                    ClientId = EastSeat.ResourceIdea.Domain.Clients.ValueObjects.ClientId.Create(ClientId.Value)
+                };
+                clientQuery.TenantId = ResourceIdeaRequestContext.Tenant;
+
+                var response = await Mediator.Send(clientQuery);
+                if (response.IsSuccess && response.Content.HasValue)
+                {
+                    Engagements = response.Content.Value;
+                }
+                else
+                {
+                    HasError = true;
+                    ErrorMessage = "Failed to load engagements for the selected client. Please try again later.";
+                }
+            }
+            else
+            {
+                // Load all engagements for the tenant
+                var query = new GetAllEngagementsQuery(1, 100);
+                query.TenantId = ResourceIdeaRequestContext.Tenant;
+
+                var response = await Mediator.Send(query);
+                if (response.IsSuccess && response.Content.HasValue)
+                {
+                    Engagements = response.Content.Value;
+                }
+                else
+                {
+                    HasError = true;
+                    ErrorMessage = "Failed to load engagements. Please try again later.";
+                }
+            }
```

Comment:
Refactor to improve the maintainability and readability by removing the if-else conditional statements.